Add example tables to 'What is vivarium good for?' (#1384)
ihmeuw/vivarium_research: docs/source/model_design/vivarium_overview/vivarium_versus_other_model_types/index.rst

- Adverse events
- Mortality rate/life expectancy 

The below tables provide a simplified illustration of what is included in a Vivarium microsimulation. Both tables are based on a microsimulation of 

> antenatal nutritional supplementation effects on birthweight and child-growth failure. 

Reviewer: ```suggestion
antenatal nutritional supplementation effects on birthweight and child growth failure. 
```

> .. image:: microsim_example_table_3.PNG

Reviewer: In addition to these images, can you include the Excel files you used to create these? That will make it much easier for people to edit in the future.

Author: Oh good point! Will do

This first table outlines some of the simulant attributes we can include in our microsimulations. The simulants listed here have demographic attributes
such as sex and location, as well as risk factors (e.g., *did the mother/birthing person received iron and folic acid (IFA) supplementation during pregnancy?*).

> .. image:: microsim_example_table_4.PNG

Reviewer: I think this is clear enough, but I do think it would be slightly clearer if these tables were split vertically for different timesteps. I was imagining a break with "... two days pass..." in between, but I'm not sure how you would do that in Excel (merged cells?). Likewise, I think a little bit of space between the two scenarios would be nice.

Author: Ah yeah, that makes sense! I can give that a try!

Author: Just tried to make those changes - lmk if they don't look quite right still!

Reviewer: Looks great to me!

This second table demonstrates our ability to simulate different intervention scenarios and compare outcomes. In this case, the baseline scenario 

> entailed no antenatal supplementation for the simulant's mother/birthing person, which led to the simulant's low birthweight. This caused the simulant

Reviewer 2: ```suggestion
entailed no antenatal supplementation for the simulant's mother/birthing person, which meant the simulant had a low birthweight. This caused the simulant
```

Simulation components are interdependent, and relationships between attributes are also based on best available real-world 
evidence. For example, disease progression is a function of simulant characteristics and treatment; survival depends on stage of disease and other characteristics. 

> As the simulation proceeds, various simulant attributes change and these changes are recorded through observers (to read more about observers, visit [insert link here]).

Reviewer: Do we have a link for this?

Author: I don't think so, which makes me wonder if there should be somewhere in the docs that explain what we mean when we say 'observer'? It's pretty intuitive but could be helpful to have something brief somewhere!

Reviewer 3: We have an open documentation ticket to add an observers page to the "types of vivarium model components." Maybe we can write a to-do note here to link to it when it is created?

Author: Oh sweet! I'll add a to-do in that case

.. note::

  Are you feeling confused by some of the vocabulary on this page? Please visit our `Vivarium glossary <https://vivarium-research.readthedocs.io/en/latest/glossary/index.html>`_ 
  for some definitions of microsimulation vocabulary!

What are the advantages and disadvantages of microsimulations?
--------------------------------------------------------------

Microsimulation, described above, is a special kind of model that falls under the broader umbrella of **individual-based modeling (IBM)**.

> In this section, we will elaborate on what differentiates microsimulation from other forms of IBM, such as agent-based simulation. 

Reviewer: As I read it, most of the pros/cons in this section are comparing microsimulation vs other types of statistical models such as compartmental models or multiplication models. I'm not sure I understand exactly the pros and cons of the way we do things vs ABM, but I think ABM can do most of these "advantages."

Advantages of microsimulation
+++++++++++++++++++++++++++++

Correlation between risk exposures
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

At the individual level, risk exposures are frequently correlated. Examples include high body mass index and high fasting plasma glucose, tobacco smoking and alcohol use, and childhood height and weight.

Usually, Vivarium assigns each risk exposure independently to simulants, such that each one follows the desired population-level univariate distribution, which frequently comes from GBD. In the common case of a dichotomous exposure, this means that each simulant (in the same age/sex/location group) has the same probability of exposure, which is equal to the prevalence in that population.

Continuous values
~~~~~~~~~~~~~~~~~

> e.g., allows the modeling of joint distribution between blood pressure and blood glucose levels -- this kind of thing is hard with categorical data characteristic of multiplication models

Reviewer: Let's have TODOs for the unfinished sections here.

Time-varying rates dependent on multiple factors
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

e.g., Multiple Myeloma model

Incorporating uncertainty
~~~~~~~~~~~~~~~~~~~~~~~~~

Our microsimulation framework is formulated to routinely propagate forward the uncertainty captured by these draws. Monte Carlo methods allow us to effectively incorporate different kinds of uncertainty into our models. 

Monte Carlo microsimulation methods provide a useful approach to modeling stochastic uncertainty in a conceptually valid manner because chance – that is, reliance on randomly generated numbers – is used to govern movement between health states for individual simulants from a single computer-generated population, as opposed to comparing the role of chance across multiple populations with differing characteristics.

> In the context of healthcare estimates typically characterized by considerable variance across quantities of interest, the use of Monte Carlo methods to run simulations using iterative draws from probability distributions provides an effective means of incorporating parameter uncertainty. Simulation methods have the added advantage of reducing confounding variance between different scenarios by ensuring that the simulated population is identical across all scenarios analyzed. The only difference is the presence or absence of health interventions. As a further refinement of these methods, microsimulations represent each person in the population as an individual simulant with attributes variably assigned across simulants to reflect the population of interest. By incorporating heterogeneity at the level of the individual simulant, microsimulations are able to provide an additional degree of flexibility and detail (such as stratifying results according to demographic or epidemiological attributes) not available in population-level models. 

Reviewer: > Simulation methods have the added advantage of reducing confounding variance between different scenarios by ensuring that the simulated population is identical across all scenarios analyzed. The only difference is the presence or absence of health interventions.

This part is about common random numbers, which as you've mentioned before aren't really an aspect of simulation methods in general but Vivarium in particular. I think this belongs in a different section.

Reviewer: I think the last part (after the part quoted above) isn't about uncertainty and could be cut or moved to another section.

> - Surfaces complications that are more hidden in non-individual-based models (e.g., compartmental models can obscure certain limitations)

Reviewer: This should be its own section under "advantages" with a TODO, I think.

Disadvantages of microsimulation
++++++++++++++++++++++++++++++++

Resource requirements
~~~~~~~~~~~~~~~~~~~~~

Our microsimulation models require a significant amount of detailed data to accurately represent individual behaviors and the relationships between them. Obtaining this data can be challenging and time-consuming, and there may be privacy concerns associated with using such detailed personal data. 

If we're not modeling the sort of complexity that microsimulation enables... perhaps a multiplication model would suit us just as well?

Residual confounding
~~~~~~~~~~~~~~~~~~~~

Getting realistic, individual-level characteristics from population-level data is difficult, and creates opportunity for residual confounding...

Simulant agency
~~~~~~~~~~~~~~~

> Unlike agent-based models, microsimulations do not necessarily program agency into individual simulants. This removes some of the real-world likeness of these models...

Reviewer: As mentioned above, this feels like the only one in this list that contrasts microsimulation with ABM. Also, full disclosure, I don't really understand it, though I know I've heard Abie say it a couple times -- do you have a reference about this to help me/others grasp it?

.. todo::

  Fill out this section with strengths and weakness of microsimulation. Could also add context about how microsim compares to agent-based simulation and the broader umbrella term of IBM.

  Explanation of relationships between macroscopic population-level measures such as incidence rate and the corresponding microscopic hazard rates we use in our simulations, and how these relationships might affect our model design and V&V.

  Cite [Allen-et-al-2019]_ and [Sorensen-et-al-2017]_ somewhere.
  

How does Vivarium compare with other microsimulation tools?
-----------------------------------------------------------

Leveraging the Global Burden of Disease (GBD) Study
+++++++++++++++++++++++++++++++++++++++++++++++++++

> Based on GBD population-level estimates stratified by age, sex, time, and location, the IHME microsimulation framework creates a simulated population that mirrors the relevant qualities of a given real-world population of interest. Each individual simulant in the population is assigned health and demographic attributes that specify characteristics such as age, sex, disease, and risk factor status (for example, diabetes diagnosis, systolic blood pressure level, or body-mass index). In any given microsimulation analysis, there can be thousands, millions, or, theoretically, billions of simulants. Although the assigned health and demographic attributes are heterogeneous across individuals within the simulated population, when taken in aggregate they match the relevant real-world population-level data supplied by the GBD.

Reviewer: What do you mean by the "IHME microsimulation framework" (and "the IHME platform" in the next paragraph)? Are these synonyms with "Vivarium"? I'd rather just say "Vivarium" if so.

The IHME platform is uniquely positioned to extract relevant input parameters from the vast repository of GBD population health data that are explicitly structured to quantify uncertainty by representing estimates as draws from Bayesian-derived joint posterior distributions.

> The IHME microsimulation platform, unlike most other microsimulation frameworks, additionally reduces variance between scenarios by appropriately controlling stochastic uncertainty through a common-random-numbers approach. In this procedure, for each simulant, the same randomly generated number is used in both baseline and intervention scenarios to dictate change or maintenance of health status across time steps.

Reviewer: I don't think this belongs in this section. I think this is a pro of Vivarium vs other individual-based modeling tools.
